src/client/Client.cc

```diff
@@ -5413,7 +5385,7 @@ int Client::authenticate()
 {
   assert(client_lock.is_locked_by_me());
 
-  if (authenticated) {
+  if (monclient->is_authenticated()) {
```

Comment:
Only at the start -- active_con can get set back to null if it loses its mon connection at some point, right?